### ISSUE TITLE
Issues on Fire HD 10 Generation 8/9 Tablets (OpenAL problem?)

### ISSUE THREAD

**Issue by Owlzy** [MonoGame/MonoGame#7011]
We have had some issues reported to us recently regarding our clients game on the Amazon App Store.  It seems this issue has started occurring fairly recently and they all report they are on a Kindle Fire HD 10 tablet (Gen 8+,  doesn't occur on our Fire HD 10 Gen 7).  

We only have fairly vague bug reports from customers, but largely they seem to say it wont launch at all.  And one or two report it either won't install, or will crash part way through loading. 

It's a paid app so people are raging pretty hard and requesting refunds.  We have a couple of these tablets on their way so hopefully we can reproduce on a device when it arrives. 

In the meantime Amazon have supplied a log of the crash.  Trawling through it I don't see a much apart from OpenAL failing to initialise.  Relevant part of logs below:

```
01-30 13:16:41.164: E/linker(18719): library "/data/app/com.pjmasks.racingheroeskindle-RbklyUwWcxwNlxByPzuN7Q==/lib/arm/libopenal32.so" ("/data/app/com.pjmasks.racingheroeskindle-RbklyUwWcxwNlxByPzuN7Q==/lib/arm/libopenal32.so") needed or dlopened by "(unknown)" is not accessible for the namespace: [name="(anonymous)", ld_library_paths="", default_library_paths="/system/lib", permitted_paths=""]
01-30 13:16:41.164: I/mono-stdout(18719): FuncLoader.LoadLibrary libopenal32.so Not Found!
01-30 13:16:41.166: E/linker(18719): library "/data/user/0/com.pjmasks.racingheroeskindle/lib/libopenal32.so" ("/data/app/com.pjmasks.racingheroeskindle-RbklyUwWcxwNlxByPzuN7Q==/lib/arm/libopenal32.so") needed or dlopened by "(unknown)" is not accessible for the namespace: [name="(anonymous)", ld_library_paths="", default_library_paths="/system/lib", permitted_paths=""]
01-30 13:16:41.167: E/linker(18719): library "/data/user/0/com.pjmasks.racingheroeskindle/lib/libopenal32.so" ("/data/app/com.pjmasks.racingheroeskindle-RbklyUwWcxwNlxByPzuN7Q==/lib/arm/libopenal32.so") needed or dlopened by "(unknown)" is not accessible for the namespace: [name="(anonymous)", ld_library_paths="", default_library_paths="/system/lib", permitted_paths=""]
01-30 13:16:41.167: I/mono-stdout(18719): FuncLoader.LoadLibrary /data/user/0/com.pjmasks.racingheroeskindle/lib/libopenal32.so Not Found!
01-30 13:16:41.172: I/MonoDroid(18719): UNHANDLED EXCEPTION:
01-30 13:16:41.192: I/MonoDroid(18719): System.DllNotFoundException: Couldn't initialize OpenAL because the native binaries couldn't be found.
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Audio.OpenALSoundController..ctor () [0x00022] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Audio.OpenALSoundController.EnsureInitialized () [0x00014] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Audio.SoundEffect.PlatformInitialize () [0x00000] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Audio.SoundEffect.Initialize () [0x0001d] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Audio.SoundEffect..ctor (System.Byte[] header, System.Byte[] buffer, System.Int32 bufferSize, System.Int32 durationMs, System.Int32 loopStart, System.Int32 loopLength) [0x00011] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Content.SoundEffectReader.Read (Microsoft.Xna.Framework.Content.ContentReader input, Microsoft.Xna.Framework.Audio.SoundEffect existingInstance) [0x00045] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Content.ContentTypeReader`1[T].Read (Microsoft.Xna.Framework.Content.ContentReader input, System.Object existingInstance) [0x00003] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Content.ContentReader.InnerReadObject[T] (T existingInstance) [0x0002d] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Content.ContentReader.ReadObject[T] () [0x00000] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Content.ContentReader.ReadAsset[T] () [0x00006] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Content.ContentManager.ReadAsset[T] (System.String assetName, System.Action`1[T] recordDisposableObject) [0x00080] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Content.ContentManager.Load[T] (System.String assetName) [0x0005a] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at P3.AssetManager.Load (System.String uri, Microsoft.Xna.Framework.Audio.SoundEffect& sound) [0x00019] in <316b730cb992480f956975de6f821770>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at P3.Game.Update (Microsoft.Xna.Framework.GameTime gameTime) [0x001a3] in <316b730cb992480f956975de6f821770>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Game.DoUpdate (Microsoft.Xna.Framework.GameTime gameTime) [0x00019] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Game.Tick () [0x00102] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.AndroidGameWindow.OnUpdateFrame (System.Object sender, Microsoft.Xna.Framework.MonoGameAndroidGameView+FrameEventArgs frameEventArgs) [0x00041] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.MonoGameAndroidGameView.UpdateFrameInternal (Microsoft.Xna.Framework.MonoGameAndroidGameView+FrameEventArgs e) [0x0000f] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.MonoGameAndroidGameView.UpdateAndRenderFrame () [0x00056] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.MonoGameAndroidGameView.processStateRunning (System.Threading.CancellationToken token) [0x0005b] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.MonoGameAndroidGameView.RunIteration (System.Threading.CancellationToken token) [0x000dd] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.MonoGameAndroidGameView+<>c__DisplayClass61_0.<WorkerThreadFrameDispatcher>b__0 (System.Object s) [0x00017] in <d7c60be6268d4de9a24a19730abb974f>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Android.App.SyncContext+<>c__DisplayClass3_0.<Send>b__0 () [0x00000] in <6a40cf997202478990245d1cdd923897>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <6a40cf997202478990245d1cdd923897>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <6a40cf997202478990245d1cdd923897>:0 
01-30 13:16:41.192: I/MonoDroid(18719):   at (wrapper dynamic-method) System.Object.18(intptr,intptr)
01-30 13:16:41.223: W/ingheroeskindl(18719): JNI RegisterNativeMethods: attempt to register 0 native methods for md5bcaacea39e485ed8e3d7f0e2edda77a7.MonoException
01-30 13:16:41.223: W/ingheroeskindl(18719): JNI RegisterNativeMethods: attempt to register 0 native methods for md5bcaacea39e485ed8e3d7f0e2edda77a7.MonoExceptionHelper
01-30 13:16:41.403: W/CrashlyticsCore(18719): Cannot send reports. Settings are unavailable.
01-30 13:16:41.453: E/AndroidRuntime(18719): FATAL EXCEPTION: main
01-30 13:16:41.453: E/AndroidRuntime(18719): Process: com.pjmasks.racingheroeskindle, PID: 18719
01-30 13:16:41.453: E/AndroidRuntime(18719): md5bcaacea39e485ed8e3d7f0e2edda77a7.MonoExceptionHelper: (DllNotFoundException) Couldn't initialize OpenAL because the native binaries couldn't be found.
01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Audio.OpenALSoundController..ctor()[0x00022](Microsoft/Xna/Framework/Audio/OpenALSoundController/0x00022.cs:1)
01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Audio.OpenALSoundController.EnsureInitialized()[0x00014](Microsoft/Xna/Framework/Audio/OpenALSoundController0x00014.cs:1)
01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Audio.SoundEffect.PlatformInitialize()[0x00000](Microsoft/Xna/Framework/Audio/SoundEffect0x00000.cs:1)
01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Audio.SoundEffect.Initialize()[0x0001d](Microsoft/Xna/Framework/Audio/SoundEffect0x0001d.cs:1)
01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Audio.SoundEffect..ctor(System.Byte[] header, System.Byte[] buffer, System.Int32 bufferSize, System.Int32 durationMs, System.Int32 loopStart, System.Int32 loopLength)[0x00011](Microsoft/Xna/Framework/Audio/SoundEffect/0x00011.cs:1)
01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Content.SoundEffectReader.Read(Microsoft.Xna.Framework.Content.ContentReader input, Microsoft.Xna.Framework.Audio.SoundEffect existingInstance)[0x00045](Microsoft/Xna/Framework/Content/SoundEffectReader0x00045.cs:1)
01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Content.ContentTypeReader`1[T].Read(Microsoft.Xna.Framework.Content.ContentReader input, System.Object existingInstance)[0x00003](Microsoft/Xna/Framework/Content/ContentTypeReader1T0x00003.cs:1)
01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Content.ContentReader.InnerReadObject[T](T existingInstance)[0x0002d](Microsoft/Xna/Framework/Content/ContentReader0x0002d.cs:1)
01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Content.ContentReader.ReadObject[T]()[0x00000](Microsoft/Xna/Framework/Content/ContentReader0x00000.cs:1)
01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Content.ContentReader.ReadAsset[T]()[0x00006](Microsoft/Xna/Framework/Content/ContentReader0x00006.cs:1)
01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Content.ContentManager.ReadAsset[T](System.String assetName, System.Action`1[T] recordDisposableObject)[0x00080](Microsoft/Xna/Framework/Content/ContentManager0x00080.cs:1)
01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Content.ContentManager.Load[T](System.String assetName)[0x0005a](Microsoft/Xna/Framework/Content/ContentManager0x0005a.cs:1)
01-30 13:16:41.453: E/AndroidRuntime(18719): 	at P3.AssetManager.Load(System.String uri, Microsoft.Xna.Framework.Audio.SoundEffect& sound)[0x00019](P3/AssetManager0x00019.cs:1)
01-30 13:16:41.453: E/AndroidRuntime(18719): 	at P3.Game.Update(Microsoft.Xna.Framework.GameTime gameTime)[0x001a3](P3/Game0x001a3.cs:1)
01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Game.DoUpdate(Microsoft.Xna.Framework.GameTime gameTime)[0x00019](Microsoft/Xna/Framework/Game0x00019.cs:1)
```

Does this seem something likely to cause the app to fail launching?  After OpenAL fails to initialise the next part of the logs shows some audio getting loaded.  I'd have to double check but from what I remember audio is the first thing we load. 

Any help would be much appreciated, got a pretty unhappy client on our hands :)

#### What version of MonoGame does the bug occur on:
- Latest Dev Version - 3.8.0.1018

#### What operating system are you using:
- Windows

#### What MonoGame platform are you using:

- Android(Fire OS)

**Comment by tomspilman** [comment]
@Owlzy - This is something we've seen as well.  I hope to have a fix soon which i'll look to submit a PR with.

**Comment by Owlzy** [comment]
Okay thanks @tomspilman , let me know if I can be of any help.

**Comment by 130133362948a** [comment]
> We have had some issues reported to us recently regarding our clients game on the Amazon App Store. It seems this issue has started occurring fairly recently and they all report they are on a Kindle Fire HD 10 tablet (Gen 8+, doesn't occur on our Fire HD 10 Gen 7).
> 
> We only have fairly vague bug reports from customers, but largely they seem to say it wont launch at all. And one or two report it either won't install, or will crash part way through loading.
> 
> It's a paid app so people are raging pretty hard and requesting refunds. We have a couple of these tablets on their way so hopefully we can reproduce on a device when it arrives.
> 
> In the meantime Amazon have supplied a log of the crash. Trawling through it I don't see a much apart from OpenAL failing to initialise. Relevant part of logs below:
> 
> ```
> 01-30 13:16:41.164: E/linker(18719): library "/data/app/com.pjmasks.racingheroeskindle-RbklyUwWcxwNlxByPzuN7Q==/lib/arm/libopenal32.so" ("/data/app/com.pjmasks.racingheroeskindle-RbklyUwWcxwNlxByPzuN7Q==/lib/arm/libopenal32.so") needed or dlopened by "(unknown)" is not accessible for the namespace: [name="(anonymous)", ld_library_paths="", default_library_paths="/system/lib", permitted_paths=""]
> 01-30 13:16:41.164: I/mono-stdout(18719): FuncLoader.LoadLibrary libopenal32.so Not Found!
> 01-30 13:16:41.166: E/linker(18719): library "/data/user/0/com.pjmasks.racingheroeskindle/lib/libopenal32.so" ("/data/app/com.pjmasks.racingheroeskindle-RbklyUwWcxwNlxByPzuN7Q==/lib/arm/libopenal32.so") needed or dlopened by "(unknown)" is not accessible for the namespace: [name="(anonymous)", ld_library_paths="", default_library_paths="/system/lib", permitted_paths=""]
> 01-30 13:16:41.167: E/linker(18719): library "/data/user/0/com.pjmasks.racingheroeskindle/lib/libopenal32.so" ("/data/app/com.pjmasks.racingheroeskindle-RbklyUwWcxwNlxByPzuN7Q==/lib/arm/libopenal32.so") needed or dlopened by "(unknown)" is not accessible for the namespace: [name="(anonymous)", ld_library_paths="", default_library_paths="/system/lib", permitted_paths=""]
> 01-30 13:16:41.167: I/mono-stdout(18719): FuncLoader.LoadLibrary /data/user/0/com.pjmasks.racingheroeskindle/lib/libopenal32.so Not Found!
> 01-30 13:16:41.172: I/MonoDroid(18719): UNHANDLED EXCEPTION:
> 01-30 13:16:41.192: I/MonoDroid(18719): System.DllNotFoundException: Couldn't initialize OpenAL because the native binaries couldn't be found.
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Audio.OpenALSoundController..ctor () [0x00022] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Audio.OpenALSoundController.EnsureInitialized () [0x00014] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Audio.SoundEffect.PlatformInitialize () [0x00000] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Audio.SoundEffect.Initialize () [0x0001d] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Audio.SoundEffect..ctor (System.Byte[] header, System.Byte[] buffer, System.Int32 bufferSize, System.Int32 durationMs, System.Int32 loopStart, System.Int32 loopLength) [0x00011] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Content.SoundEffectReader.Read (Microsoft.Xna.Framework.Content.ContentReader input, Microsoft.Xna.Framework.Audio.SoundEffect existingInstance) [0x00045] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Content.ContentTypeReader`1[T].Read (Microsoft.Xna.Framework.Content.ContentReader input, System.Object existingInstance) [0x00003] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Content.ContentReader.InnerReadObject[T] (T existingInstance) [0x0002d] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Content.ContentReader.ReadObject[T] () [0x00000] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Content.ContentReader.ReadAsset[T] () [0x00006] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Content.ContentManager.ReadAsset[T] (System.String assetName, System.Action`1[T] recordDisposableObject) [0x00080] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Content.ContentManager.Load[T] (System.String assetName) [0x0005a] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at P3.AssetManager.Load (System.String uri, Microsoft.Xna.Framework.Audio.SoundEffect& sound) [0x00019] in <316b730cb992480f956975de6f821770>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at P3.Game.Update (Microsoft.Xna.Framework.GameTime gameTime) [0x001a3] in <316b730cb992480f956975de6f821770>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Game.DoUpdate (Microsoft.Xna.Framework.GameTime gameTime) [0x00019] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.Game.Tick () [0x00102] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.AndroidGameWindow.OnUpdateFrame (System.Object sender, Microsoft.Xna.Framework.MonoGameAndroidGameView+FrameEventArgs frameEventArgs) [0x00041] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.MonoGameAndroidGameView.UpdateFrameInternal (Microsoft.Xna.Framework.MonoGameAndroidGameView+FrameEventArgs e) [0x0000f] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.MonoGameAndroidGameView.UpdateAndRenderFrame () [0x00056] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.MonoGameAndroidGameView.processStateRunning (System.Threading.CancellationToken token) [0x0005b] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.MonoGameAndroidGameView.RunIteration (System.Threading.CancellationToken token) [0x000dd] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Microsoft.Xna.Framework.MonoGameAndroidGameView+<>c__DisplayClass61_0.<WorkerThreadFrameDispatcher>b__0 (System.Object s) [0x00017] in <d7c60be6268d4de9a24a19730abb974f>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Android.App.SyncContext+<>c__DisplayClass3_0.<Send>b__0 () [0x00000] in <6a40cf997202478990245d1cdd923897>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <6a40cf997202478990245d1cdd923897>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <6a40cf997202478990245d1cdd923897>:0 
> 01-30 13:16:41.192: I/MonoDroid(18719):   at (wrapper dynamic-method) System.Object.18(intptr,intptr)
> 01-30 13:16:41.223: W/ingheroeskindl(18719): JNI RegisterNativeMethods: attempt to register 0 native methods for md5bcaacea39e485ed8e3d7f0e2edda77a7.MonoException
> 01-30 13:16:41.223: W/ingheroeskindl(18719): JNI RegisterNativeMethods: attempt to register 0 native methods for md5bcaacea39e485ed8e3d7f0e2edda77a7.MonoExceptionHelper
> 01-30 13:16:41.403: W/CrashlyticsCore(18719): Cannot send reports. Settings are unavailable.
> 01-30 13:16:41.453: E/AndroidRuntime(18719): FATAL EXCEPTION: main
> 01-30 13:16:41.453: E/AndroidRuntime(18719): Process: com.pjmasks.racingheroeskindle, PID: 18719
> 01-30 13:16:41.453: E/AndroidRuntime(18719): md5bcaacea39e485ed8e3d7f0e2edda77a7.MonoExceptionHelper: (DllNotFoundException) Couldn't initialize OpenAL because the native binaries couldn't be found.
> 01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Audio.OpenALSoundController..ctor()[0x00022](Microsoft/Xna/Framework/Audio/OpenALSoundController/0x00022.cs:1)
> 01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Audio.OpenALSoundController.EnsureInitialized()[0x00014](Microsoft/Xna/Framework/Audio/OpenALSoundController0x00014.cs:1)
> 01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Audio.SoundEffect.PlatformInitialize()[0x00000](Microsoft/Xna/Framework/Audio/SoundEffect0x00000.cs:1)
> 01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Audio.SoundEffect.Initialize()[0x0001d](Microsoft/Xna/Framework/Audio/SoundEffect0x0001d.cs:1)
> 01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Audio.SoundEffect..ctor(System.Byte[] header, System.Byte[] buffer, System.Int32 bufferSize, System.Int32 durationMs, System.Int32 loopStart, System.Int32 loopLength)[0x00011](Microsoft/Xna/Framework/Audio/SoundEffect/0x00011.cs:1)
> 01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Content.SoundEffectReader.Read(Microsoft.Xna.Framework.Content.ContentReader input, Microsoft.Xna.Framework.Audio.SoundEffect existingInstance)[0x00045](Microsoft/Xna/Framework/Content/SoundEffectReader0x00045.cs:1)
> 01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Content.ContentTypeReader`1[T].Read(Microsoft.Xna.Framework.Content.ContentReader input, System.Object existingInstance)[0x00003](Microsoft/Xna/Framework/Content/ContentTypeReader1T0x00003.cs:1)
> 01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Content.ContentReader.InnerReadObject[T](T existingInstance)[0x0002d](Microsoft/Xna/Framework/Content/ContentReader0x0002d.cs:1)
> 01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Content.ContentReader.ReadObject[T]()[0x00000](Microsoft/Xna/Framework/Content/ContentReader0x00000.cs:1)
> 01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Content.ContentReader.ReadAsset[T]()[0x00006](Microsoft/Xna/Framework/Content/ContentReader0x00006.cs:1)
> 01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Content.ContentManager.ReadAsset[T](System.String assetName, System.Action`1[T] recordDisposableObject)[0x00080](Microsoft/Xna/Framework/Content/ContentManager0x00080.cs:1)
> 01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Content.ContentManager.Load[T](System.String assetName)[0x0005a](Microsoft/Xna/Framework/Content/ContentManager0x0005a.cs:1)
> 01-30 13:16:41.453: E/AndroidRuntime(18719): 	at P3.AssetManager.Load(System.String uri, Microsoft.Xna.Framework.Audio.SoundEffect& sound)[0x00019](P3/AssetManager0x00019.cs:1)
> 01-30 13:16:41.453: E/AndroidRuntime(18719): 	at P3.Game.Update(Microsoft.Xna.Framework.GameTime gameTime)[0x001a3](P3/Game0x001a3.cs:1)
> 01-30 13:16:41.453: E/AndroidRuntime(18719): 	at Microsoft.Xna.Framework.Game.DoUpdate(Microsoft.Xna.Framework.GameTime gameTime)[0x00019](Microsoft/Xna/Framework/Game0x00019.cs:1)
> ```
> 
> Does this seem something likely to cause the app to fail launching? After OpenAL fails to initialise the next part of the logs shows some audio getting loaded. I'd have to double check but from what I remember audio is the first thing we load.
> 
> Any help would be much appreciated, got a pretty unhappy client on our hands :)
> 
> #### What version of MonoGame does the bug occur on:
> * Latest Dev Version - 3.8.0.1018
> 
> #### What operating system are you using:
> * Windows
> 
> #### What MonoGame platform are you using:
> * Android(Fire OS)

Fukoff

**Comment by Owlzy** [comment]
I think this issue can be closed, for me it no longer occurs.  I didn't do anything specifically to fix it, but I did recently install Visual Studio 2019, so I'm guessing that what caused it to suddenly work. 

Info from my VS studio about :

```
Microsoft Visual Studio Community 2019
Version 16.4.5
VisualStudio.16.Release/16.4.5+29806.167
Microsoft .NET Framework
Version 4.8.03752

Installed Version: Community

Azure App Service Tools v3.0.0   16.4.460.23317
Azure App Service Tools v3.0.0

C# Tools   3.4.1-beta4-19614-01+165046097562cfe65b09c2e9a9d8f7cd88526f2c
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.0 (d16-2@8b56e20)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Mono Debugging for Visual Studio   16.5.24 (1fafd7e)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   5.4.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Visual Basic Tools   3.4.1-beta4-19614-01+165046097562cfe65b09c2e9a9d8f7cd88526f2c
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools 10.4 for F# 4.6   16.4.0-beta.19556.5+e7597deb7042710a7142bdccabd6f92b0840d354
Microsoft Visual F# Tools 10.4 for F# 4.6

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Foo   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Xamarin   16.4.000.311 (d16-4@ddfd842)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   16.4.0.475 (remotes/origin/d16-4@ac250f5aa)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   16.4.25 (579ee62)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   10.1.4.0 (d16-4/e44d1ae)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: fd9f379
    Java.Interop: xamarin/java.interop/d16-4@c4e569f
    ProGuard: xamarin/proguard/master@905836d
    SQLite: xamarin/sqlite/3.28.0@46204c4
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-5@9f4ed4b


Xamarin.iOS and Xamarin.Mac SDK   13.10.0.17 (5f802ef)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.

Xenko.VisualStudio.Package   2.0
Xenko VisualStudio Package
```

**Comment by Owlzy** [comment]
Am closing, have tested and confirmed on a ton of devices and the issue is no longer present for us.  Seems just need to make sure your up to date on your Xamarin version, we also had a heap of problems with iOS which are now solved simply by upgrading VStudio / XCode / Xamarin.

**Comment by tomspilman** [comment]
FYI.

This issue was fixed by Fire OS 7.3.1.2 which is the latest version of Fire OS as of the time of this post.  It seems the bug occured on 7.3.1.1 or earlier.

Unless there is some other issue it seems like telling users to update the OS on their tablet is the right fix.
